doc/source/whatsnew/v1.4.0.rst

```diff
@@ -258,7 +258,7 @@ I/O
 - Bug in :func:`read_excel` attempting to read chart sheets from .xlsx files (:issue:`41448`)
 - Bug in :func:`json_normalize` where ``errors=ignore`` could fail to ignore missing values of ``meta`` when ``record_path`` has a length greater than one (:issue:`41876`)
 - Bug in :func:`read_csv` with multi-header input and arguments referencing column names as tuples (:issue:`42446`)
--
+- Bug in :func:`read_fwf` where difference in length of ``colspecs`` and ``names`` did not throw any error (:issue:`40830`)
```

Comment:
Maybe be more specific as to the type of error now being thrown, for example "Bug in ... not throwing ``ValueError``"